### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by lewinther**
Next step in the project is to implement the API and server for the logic specifically made in the News section to be functional. We are planning to make similar logic in Message and profile sections, where the context throughout the site will be that of the specific user (based on their docking number).

### CODE REVIEW

File: src/assets/Icons_DB-home.svg
Comment (mircealungu) [inline src/assets/Icons_DB-home.svg:1]: this is reduntant?

File: src/assets/Icons_DB-inbox.svg
Comment (mircealungu) [inline src/assets/Icons_DB-inbox.svg:1]: ditto?

File: src/assets/Message_BoatPicture.webp
Comment (mircealungu) [inline src/assets/Message_BoatPicture.webp:1]: what's a .webp extension?

File: src/components/DockFilter.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+  const handleChange = (e) => {
+    const newDockNumber = e.target.value;
+    setDockNumber(newDockNumber);
+    handleSelectionChange(newDockNumber);
+  };
+
+  return (
+    <form className="dock-select-form" onSubmit={(e) => e.preventDefault()}>
+      <label htmlFor="dock-select" className="dock-select-label">
```
Comment (mircealungu) [inline src/components/DockFilter.js:18]: can you simply use `for`? https://github.com/itu-tid/lecture-notes/blob/main/Tech-TopUps/5-Coding-Practices.md#associate-labels-with-inputs-explicitly-with-for

File: src/components/DockFilterTest.js
Comment (mircealungu) [inline src/components/DockFilterTest.js:1]: maybe move to a testing folder?

File: src/components/InputMessage.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { Fragment } from "react";
+
+export default function InputMessage(){
+
+    return (
+        <Fragment>
```
Comment (mircealungu) [inline src/components/InputMessage.js:6]: try to use a automatic formatter.

File: src/components/MessageCardContainer.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import { Fragment } from "react";
+
+import MessageCard from "./MessageCard";
+
+export default function MessageCardContainer(){
+
+	return(
+		<Fragment>
+			<h3 className="centered"  style={{ marginTop: '30px' }}  >Check your inbox:</h3>
```
Comment (mircealungu) [inline src/components/MessageCardContainer.js:9]: don't define styles inline. but do it in the CSS

File: src/components/MessageCardContainerINBOX.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import { Fragment } from "react";
+import { Link } from "react-router-dom";
+
+import "../../src/styles.css";
+import MessageCard from "./MessageCard";
+
+export default function MessageCardContainerINBOX(){
+	return(
+		<Fragment>
```
Comment (mircealungu) [inline src/components/MessageCardContainerINBOX.js:9]: this is really ugly formatting

File: src/components/NavbarBottom.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+            </Link>
+            <Link className="link" to={`/Inbox`}>
+              <div className={`nav-item ${activeItem === 'Inbox' ? 'active' : ''}`} onClick={() => handleItemClick("Inbox")}>
+                  <IconInbox className="icon" />
+                  <p>Inbox</p>
+              </div>
+            </Link>
+            <Link className="link" to={`/NewMessage`}>
+              <div className={`nav-item ${activeItem === 'Newm' ? 'active' : ''}`} onClick={() => handleItemClick("Newm")}>
+                  <IconNewm className="icon" />
```
Comment (mircealungu) [inline src/components/NavbarBottom.js:36]: IconNewMessage ...?

File: src/components/News.js [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+export default function News({
+    newsImg,
+    newsTitle,
+    newsDate,
+    newsTekst,
+}){
+
+    const { newsId: newsId } = useParams();
+
+    const getNewsData = async () => {
```
Comment (mircealungu) [inline src/components/News.js:14]: dead code?

File: src/components/SearchBarTestApp.js
Comment (mircealungu) [inline src/components/SearchBarTestApp.js:1]: ditto

File: src/components/UserInfo.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+    userName,
+    userAddress,
+    userPhone,
+    userEmail
+}) {
+
+    return(
+        <Fragment>
+            <section className="news-card-container in-column">
+                <p className="in-line">Name: {userName}</p>
```
Comment (mircealungu) [inline src/components/UserInfo.js:13]: maybe define the style only once for all the `p` elements inside of the `news-card-container`
```
.news-card-container > p { 
}```

**Review by mircealungu — CHANGES_REQUESTED** [review]
good overall but please look at the observations

**Comment by mircealungu** [comment]
@lewinther - don't forget to tag me once you're done

File: src/components/DockFilter.js (hunk 1 above)
Comment (lewinther) [reply src/components/DockFilter.js:18]: That's a great observation, it does work.

File: src/components/MessageCardContainerINBOX.js (hunk 4 above)
Comment (lewinther) [reply src/components/MessageCardContainerINBOX.js]: Thank you so much, we have prettified it.

File: src/components/NavbarBottom.js (hunk 5 above)
Comment (lewinther) [reply src/components/NavbarBottom.js]: Yes!

File: src/components/MessageCardContainer.js (hunk 3 above)
Comment (lewinther) [reply src/components/MessageCardContainer.js]: We've created a margin-top class in CSS

File: src/components/News.js (hunk 6 above)
Comment (lewinther) [reply src/components/News.js]: We might end up need it later on, if we end up having a page for news

File: src/components/UserInfo.js (hunk 7 above)
Comment (lewinther) [reply src/components/UserInfo.js]: This component is not yet completed, we'll fix the definitions in later iterations

**Comment by lewinther** [comment]
@mircealungu we have gone through the feedback notes and answered the ones we still have doubts about.

File: src/assets/Message_BoatPicture.webp
Comment (mircealungu) [reply src/assets/Message_BoatPicture.webp:1]: Thanks! Can you share the link also on LearnIt? :)

**Comment by mircealungu** [comment]
> @mircealungu we have gone through the feedback notes and answered the ones we still have doubts about.

I don't think I've seen any new comments. Is there maybe a problem with the GH UI? Can you paste below an example snapshot of a comment that you answered?

File: src/components/News.js (hunk 6 above)
Comment (lewinther) [reply src/components/News.js]: @mircealungu this is an example of a comment.

File: src/components/UserInfo.js (hunk 7 above)
Comment (lewinther) [reply src/components/UserInfo.js]: @mircealungu also a comment.

**Comment by lewinther** [comment]
I have tagged you @mircealungu in two comments

**Comment by mircealungu** [comment]
Thanks!

Everything is collapsed:

￼

However, indeed, if I expand then I find comments in some of these collapsed files:

￼

It’s like opening kinder surprise toys. Will it contain a comment? Will it not? 

But I think I understand now: 
- when you mark it as resolved, then it does not show it to me unless I expand it.

So life lesson: 
- if you want me to see things, don’t mark as resolved. 

And in your case, given that you’ve marked everything as resolved, I’ll consider that all is well and approve your merge.
Unless there’s something else that you want to talk about?

M.

**Comment by lewinther** [comment]
We love Kinder surprises! 
But for the future will remember not to mark as resolved, if we want further commentary. ;)

**Review by mircealungu — APPROVED** [review]
@igawaclawska - from my POV, they are ready to merge.

**Pull request merged**